internal/cli/runner_inspect.go

```diff
@@ -0,0 +1,136 @@
+package cli
+
+import (
+	"fmt"
+	"github.com/dustin/go-humanize"
+	"github.com/golang/protobuf/ptypes"
+	"reflect"
+	"strings"
+
+	"github.com/golang/protobuf/jsonpb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"github.com/posener/complete"
+)
+
+type RunnerInspectCommand struct {
+	*baseCommand
+
+	flagJson bool
+}
+
+func (c *RunnerInspectCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoLocalServer(), // no auth in local mode
+		WithNoConfig(),
+	); err != nil {
+		return 1
+	}
+
+	if len(c.args) == 0 {
+		c.ui.Output("Runner name required.", terminal.WithErrorStyle())
+		return 1
+	}
+	id := c.args[0]
+
+	resp, err := c.project.Client().GetRunner(c.Ctx, &pb.GetRunnerRequest{RunnerId: id})
+	if err != nil && status.Code(err) != codes.NotFound {
+		c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+		return 1
+	}
+
+	if c.flagJson {
+		var m jsonpb.Marshaler
+		m.Indent = "\t"
+		str, err := m.MarshalToString(resp)
+		if err != nil {
+			c.ui.Output(clierrors.Humanize(err), terminal.WithErrorStyle())
+			return 1
+		}
+
+		fmt.Println(str)
+		return 0
+	}
+
+	kindMap := map[reflect.Type]string{
+		reflect.TypeOf((*pb.Runner_Odr)(nil)):     "on-demand",
+		reflect.TypeOf((*pb.Runner_Local_)(nil)):  "local",
+		reflect.TypeOf((*pb.Runner_Remote_)(nil)): "remote",
+	}
+	var kindStr = "unknown"
+	if v, ok := kindMap[reflect.TypeOf(resp.Kind)]; ok {
+		kindStr = v
+	}
+	var lastSeenStr string
+	if v, err := ptypes.Timestamp(resp.LastSeen); err == nil {
+		lastSeenStr = humanize.Time(v)
+	}
+	var stateStr = "unknown"
+	stateStr = strings.ToLower(resp.AdoptionState.String())
```

Comment:
Hm, I think we can simplify this with a `switch`?

```
switch kind := resp.Kind.(type) {
case *pb.Runner_Odr:
case ....
...
}
```

Reply:
seems this was based off of https://github.com/hashicorp/waypoint/blob/357e3d3f9f4189e5abfce09253f3e14f8f362ed4/internal/cli/runner_list.go#L62-L102 from `runner_list`, so it's at least consistent with that command

Reply:
Ah yeah, I remember seeing that go by now. I think maybe switch is easier to read if it can be done? Not a huge deal